midi_app_controller/controller/_tests/test_connected_controller.py

```diff
@@ -0,0 +1,316 @@
+    thread.wait()
+
+
+def test_init_buttons(connected_controller):
+    connected_controller.turn_off_button_led = MagicMock()
+    connected_controller.init_buttons()
+    assert connected_controller.turn_off_button_led.call_count == 3
+
+
+def test_init_knobs(connected_controller):
+    connected_controller.change_knob_value = MagicMock()
+    connected_controller.init_knobs()
+    assert connected_controller.change_knob_value.call_count == 3
+
+
+@pytest.mark.parametrize("paused", [True, False])
+def test_handle_button_disengagement(paused, connected_controller):
+    connected_controller.paused = paused
+    connected_controller.actions_handler.handle_button_action = MagicMock()
+    connected_controller.paused_button_callback = MagicMock()
+    connected_controller.handle_button_disengagement([1])
+
+    if not paused:
+        connected_controller.actions_handler.handle_button_action.assert_called_once_with(
+            1
+        )
+        connected_controller.paused_button_callback.assert_not_called()
+    else:
+        connected_controller.actions_handler.handle_button_action.assert_not_called()
+        connected_controller.paused_button_callback.assert_called_once_with(1)
+
+
+@pytest.mark.parametrize("paused", [True, False])
+def test_handle_knob_message(paused, connected_controller):
+    connected_controller.actions_handler.handle_knob_action = MagicMock()
+    connected_controller.paused_knob_callback = MagicMock()
+    connected_controller.paused = paused
+    connected_controller.knob_engagement[1] = 0
+    connected_controller.handle_knob_message([1, 64])
+    if not paused:
+        connected_controller.actions_handler.handle_knob_action.assert_called_once_with(
+            knob_id=1, old_value=0, new_value=64
+        )
+        connected_controller.paused_knob_callback.assert_not_called()
+    else:
+        connected_controller.actions_handler.handle_knob_action.assert_not_called()
+        connected_controller.paused_knob_callback.assert_called_once_with(1)
+
+
+def test_send_midi_message(connected_controller):
+    connected_controller.midi_out.send_message = MagicMock(side_effect=ValueError)
+    with patch("logging.error") as mock_logging_error:
+        connected_controller.send_midi_message([1, 2, 3])
+        connected_controller.midi_out.send_message.assert_called_once_with([1, 2, 3])
+        mock_logging_error.assert_called_once()
+
+
+def test_check_set_and_run():
+    func = MagicMock()
+    ConnectedController.check_set_and_run(func, 1, QMutex(), set())
+    func.assert_called_once()
+
+
+def test_build_message(connected_controller):
+    message = connected_controller.build_message(129, 1, [2, 3])
+    assert message == [129, 2, 3]
+
+
+def test_change_knob_value(connected_controller):
+    connected_controller.send_midi_message = MagicMock()
+    connected_controller.change_knob_value(1, 129)
+    connected_controller.send_midi_message.assert_called_with([186, 1, 129])
```

Comment:
The test works but knobs should have values in the range `[0, 127]`.
```suggestion
    connected_controller.change_knob_value(1, 100)
    connected_controller.send_midi_message.assert_called_with([186, 1, 100])
```